### PR TITLE
Full examples bcic iv 2a

### PR DESCRIPTION
**Pull request by robintibor**
Examples for BCIC IV 2a with good accuracies.

So, `plot_bcic_iv_2a_moabb_trial.py` and `plot_bcic_iv_2a_moabb_cropped.py`
should replace:
* plot_bcic_iv_2a_moabb.py
* bcic_iv_2a_cropped.py
* bcic_iv_2a_trial.py

Here are results from 3 seeds, 800 epochs for cropped, 1600 epochs for trial, averaged over the 9 subjects. This cosine-annealing-train-on-full-train setup outperforms our previous early-stopping setup as is expected, previous setup was also fairly bad results in that run, don't know why unfortunately. The accuracies for a cropped deep on 0 Hz are a bit suspiciously high ? Or is this in the range you see @sbbrandt ? Trialwise is very bad, indicating the hyperparameters (learning rate, weight decay, n_epochs) are not well-suited for trialwise decoding, at least on this dataset.

skorch_braindecode_bcic_iv_2a

### CODE REVIEW

**Comment by robintibor** [comment]
You can also have a look @sliwy  I put them into new files so we don't have too many conflicts with your PR atm

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import mne
+from skorch.callbacks import LRScheduler
+mne.set_log_level('ERROR')
+
+from braindecode.datautil.windowers import create_windows_from_events
+from braindecode.classifier import EEGClassifier
+from braindecode.datasets import MOABBDataset
+from braindecode.losses import CroppedNLLLoss
+from braindecode.models.deep4 import Deep4Net
+from braindecode.models.shallow_fbcsp import ShallowFBCSPNet
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_cropped.py]:
```suggestion
from braindecode.models import ShallowFBCSPNet
```

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import torch
+import mne
+from skorch.callbacks import LRScheduler
+mne.set_log_level('ERROR')
+
+from braindecode.datautil.windowers import create_windows_from_events
+from braindecode.classifier import EEGClassifier
+from braindecode.datasets import MOABBDataset
+from braindecode.losses import CroppedNLLLoss
+from braindecode.models.deep4 import Deep4Net
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_cropped.py]:
```suggestion
from braindecode.models import Deep4Net
```

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+from functools import partial
+
+import numpy as np
+import torch
+import mne
+from skorch.callbacks import LRScheduler
+mne.set_log_level('ERROR')
+
+from braindecode.datautil.windowers import create_windows_from_events
+from braindecode.classifier import EEGClassifier
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_cropped.py]:
```suggestion
from braindecode import EEGClassifier
```

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+from collections import OrderedDict
+from functools import partial
+
+import numpy as np
+import torch
+import mne
+from skorch.callbacks import LRScheduler
+mne.set_log_level('ERROR')
+
+from braindecode.datautil.windowers import create_windows_from_events
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_cropped.py]:
```suggestion
from braindecode.datautil import create_windows_from_events
```
let's avoid 3 layers of nesting for import of this one

File: examples/plot_bcic_iv_2a_moabb_cropped.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+from braindecode.classifier import EEGClassifier
+from braindecode.datasets import MOABBDataset
+from braindecode.losses import CroppedNLLLoss
+from braindecode.models.deep4 import Deep4Net
+from braindecode.models.shallow_fbcsp import ShallowFBCSPNet
+from braindecode.models.util import to_dense_prediction_model, get_output_shape
+from braindecode.util import set_random_seeds
+from braindecode.datautil.signalproc import exponential_running_standardize
+from braindecode.datautil.transforms import transform_concat_ds
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_cropped.py]:
```suggestion
from braindecode.datautil.transforms import concat_datasets
```

red flag is the fact that transform is repeated. The module name is part of the naming of the function

File: examples/plot_bcic_iv_2a_moabb_trial.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
+"""
+Trialwise Decoding on BCIC IV 2a Competition Set with skorch and moabb.
+=====================================================================
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_trial.py]:
```suggestion
=======================================================================
```

File: examples/plot_bcic_iv_2a_moabb_trial.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+mne.set_log_level('ERROR')
+
+from braindecode.datautil.windowers import create_windows_from_events
+from braindecode.classifier import EEGClassifier
+from braindecode.datasets import MOABBDataset
+from braindecode.models.deep4 import Deep4Net
+from braindecode.models.shallow_fbcsp import ShallowFBCSPNet
+from braindecode.util import set_random_seeds
+from braindecode.datautil.signalproc import exponential_running_standardize
+from braindecode.datautil.transforms import transform_concat_ds
```
Comment (agramfort) [inline examples/plot_bcic_iv_2a_moabb_trial.py]: same remarks for imports here

File: examples/plot_bcic_iv_2a_moabb_cropped.py (hunk 5 above)
Comment (robintibor) [reply examples/plot_bcic_iv_2a_moabb_cropped.py]: hmhm, but here I think this suggestion would be highly confusing, like when you would call this function, e.g.:
`concat_datasets([list of functions to be applied])` ... 
it would be superconfusing to me (as it does not concat datasets but applies transforms to each dataset in a concatenated dataset).

**Comment by agramfort** [comment]
you realize that I am confuse about what "transform_concat_ds" means and
does :)

>

**Comment by agramfort** [comment]
CI work like a charm https://299-232335424-gh.circle-artifacts.com/0/dev/auto_examples/plot_bcic_iv_2a_moabb_cropped.html#sphx-glr-auto-examples-plot-bcic-iv-2a-moabb-cropped-py :)

**Comment by robintibor** [comment]
pictures for you @agramfort  https://302-232335424-gh.circle-artifacts.com/0/dev/auto_examples/index.html ;)

**Comment by agramfort** [comment]
now I see nice plots https://braindecode.org/auto_examples/index.html :)

not sure why I don't see any xlabel though

>

**Comment by robintibor** [comment]
thanks for the catch, xlabel should become visible in https://github.com/braindecode/braindecode/pull/94 needed to call `plt.tight_layout()`
